src/oracleFactory/updateOracle.ts

```diff
@@ -174,24 +187,26 @@ const updateOracle = ({
           safeObserver.next({
             message: 'FETCH_DATASET_ORDER',
           });
-          const datasetorderbook = await iexec.orderbook
-            .fetchDatasetOrderbook(datasetAddress, {
+          const datasetorderbook = await iexec.orderbook.fetchDatasetOrderbook(
+            datasetAddress,
+            {
               minTag: ['tee', 'scone'],
               maxTag: ['tee', 'scone'],
               requester: await iexec.wallet.getAddress(),
               workerpool,
               app: ORACLE_APP_ADDRESS,
-            })
-            .catch((e) => {
-              throw new WorkflowError('Failed to fetch datasetorder', e);
-            });
+            }
+          );
           if (abort) return;
           datasetorder =
             datasetorderbook &&
             datasetorderbook.orders[0] &&
             datasetorderbook.orders[0].order;
           if (!datasetorder) {
-            throw new WorkflowError('No datasetorder published');
+            throw new WorkflowError({
+              message: updateErrorMessage,
+              errorCause: Error('No dataset order published'),
```

Comment:
```suggestion
              errorCause: 'No dataset order published',
```
no need to wrap it into error